samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/servlet/cp/QuestionPoolStatisticsServlet.java

```diff
@@ -0,0 +1,161 @@
+import org.springframework.web.util.UriComponentsBuilder;
+
+import com.google.gson.Gson;
+
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class QuestionPoolStatisticsServlet extends SamigoBaseServlet {
+
+
+    public static final String PARAM_SITE_ID = "siteId";
+    public static final String PARAM_QUESTION_POOL_ID = "qpId";
+
+    private static final ResourceLoader EVALUATION_BUNDLE = new ResourceLoader(SamigoConstants.EVAL_BUNDLE);
+
+    private String naMessage = null;
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        // Required
+        String questionPoolId = StringUtils.trimToNull(request.getParameter(PARAM_QUESTION_POOL_ID));
+        String siteId = StringUtils.trimToNull(request.getParameter(PARAM_SITE_ID));
+
+        // Check if required params are present
+        if (StringUtils.isAnyBlank(siteId, questionPoolId) || !NumberUtils.isParsable(questionPoolId)) {
+            response.sendError(HttpServletResponse.SC_BAD_REQUEST,
+                    String.format("At least one required param is not valid. %s=[%s], %s=[%s]",
+                            PARAM_SITE_ID, siteId, PARAM_QUESTION_POOL_ID, questionPoolId));
+            return;
+        }
+
+        Optional<String> optUserId = getUserId();
+
+        // Check permissions
+        if (!canGetStatistics(siteId)) {
+            optUserId.ifPresentOrElse(
+                userId -> log.warn("User with id [{}] is not allowed to get statistics for question pool with id [{}] on site [{}]",
+                        userId, questionPoolId, siteId),
+                () -> log.warn("Unauthenticated user tried to get statistics for question pool with id [{}] on site [{}]",
+                        questionPoolId, siteId)
+            );
+
+            response.sendError(HttpServletResponse.SC_FORBIDDEN);
+            return;
+        }
+
+        String userId = optUserId.get();
+
+        // Get assessment
+        QuestionPoolService questionPoolService = new QuestionPoolService();
+
+        QuestionPoolFacade questionPool = questionPoolService.getPool(Long.valueOf(questionPoolId), userId);
+
+        // Check if questionPool was found
+        if (questionPool == null) {
+            response.sendError(HttpServletResponse.SC_BAD_REQUEST,
+                    String.format("No questionPool with id [%s] found. userId=[%s]", questionPoolId, userId));
+            return;
+        }
+
+        // Set headers
+        response.setContentType(CONTENT_TYPE_JSON);
+
+        StatisticsService statisticsService = new StatisticsService();
+        QuestionPoolStatistics qpStatistics = statisticsService.getQuestionPoolStatistics(Long.valueOf(questionPoolId));
+
+        PrintWriter out = response.getWriter();
+        writeJson(out, dataMap(qpStatistics));
+        out.flush();
+        out.close();
+    }
+
+    public static String url(String questionPoolId, String siteId) {
```

Comment:
Just curious about this method? What is it for? Could it be renamed to get more info?

Reply:
It's kind of a factory method to get the url to access the endpoint. It's intended to be used in a bean. I will see if it's still in use, or not anymore, then I'll rename or remove it.

Reply:
removed